owlkettle/guidsl.nim

```diff
@@ -193,42 +192,33 @@ proc gen(node: Node, stmts, parent: NimNode) =
       else:
         stmts.add(name)
     of NodeAttribute:
-      stmts.add(new_assignment(
-        new_dot_expr(parent, "has_" & node.name),
-        bind_sym("true")
-      ))
-      stmts.add(new_assignment(
-        new_dot_expr(parent, "val_" & node.name),
-        node.value
-      ))
+      stmts.add:
+        new_assignment(new_dot_expr(parent, "has_" & node.name), bind_sym("true"))
+      stmts.add:
+        new_assignment(new_dot_expr(parent, "val_" & node.name), node.value)
```

Comment:
Maybe use `gen_ast` here as well? I think that it is more readable that way (untested):
```nim
gen_ast(parent, has = ident("has_" & node.name), val = ident("val_" & node.name), value = node.value):
  parent.has = true
  parent.val = value
```